PWGHF/TableProducer/candidateCreatorB0.cxx

```diff
@@ -120,216 +120,143 @@ struct HfCandidateCreatorB0 {
-      auto trackParVarD = o2::dataformats::V0(df3.getPCACandidatePos(), pVecD, df3.calcPCACovMatrixFlat(),
-                                              trackParVarPiK, trackParVar2, {0, 0}, {0, 0});
-
-      int index0D = dCand.prong0Id();
-      int index1D = dCand.prong1Id();
-      int index2D = dCand.prong2Id();
-
-      // loop on D-
-      // D- → π- K+ π-
-      // we don't have direct access to D sign so we use the sign of the daughters (the pion track0 here)
-      if (track0.sign() < 0) {
-        // loop over pions
-        for (auto const& trackPion : tracksPion) {
-          // minimum pT selection
-          if (trackPion.pt() < ptPionMin || !isSelectedTrackDCA(trackPion)) {
-            continue;
-          }
-          // we reject pions that are D daughters
-          if (trackPion.globalIndex() == index0D || trackPion.globalIndex() == index1D || trackPion.globalIndex() == index2D) {
-            continue;
-          }
-          // we only keep pi+ to combine them with Dminus and reconstruct B0
-          if (trackPion.sign() < 0) {
-            continue;
-          }
-
-          hPtPion->Fill(trackPion.pt());
-          array<float, 3> pVecPion;
-          auto trackParVarPi = getTrackParCov(trackPion);
-
-          // ---------------------------------
-          // reconstruct the 2-prong B0 vertex
-          if (df2.process(trackParVarD, trackParVarPi) == 0) {
-            continue;
-          }
-
-          // calculate invariant mass
-          massDPi = RecoDecay::m(array{pVecD, pVecPion}, array{massD, massPi});
-          if (std::abs(massDPi - massB0) > invMassWindowB0) {
-            continue;
-          }
+      auto trackParCovPiK = o2::dataformats::V0(df3.getPCACandidatePos(), pVecpiK, df3.calcPCACovMatrixFlat(),
+                                                trackParCov0, trackParCov1, {0, 0}, {0, 0});
+      auto trackParCovD = o2::dataformats::V0(df3.getPCACandidatePos(), pVecD, df3.calcPCACovMatrixFlat(),
+                                              trackParCovPiK, trackParCov2, {0, 0}, {0, 0});
+
+      int index0D = track0.globalIndex();
+      int index1D = track1.globalIndex();
+      int index2D = track2.globalIndex();
+
+      // loop over pions
+      for (const auto& trackPion : tracksPion) {
+        // minimum pT selection
+        if (trackPion.pt() < ptPionMin || !isSelectedTrackDCA(trackPion)) {
+          continue;
+        }
+        // reject pions that are D daughters
+        if (trackPion.globalIndex() == index0D || trackPion.globalIndex() == index1D || trackPion.globalIndex() == index2D) {
+          continue;
+        }
+        // reject pi and D with same sign
+        if (trackPion.sign() * track0.sign() > 0) {
+          continue;
+        }
 
-          hMassB0ToDPi->Fill(massDPi);
+        hPtPion->Fill(trackPion.pt());
+        array<float, 3> pVecPion = {trackPion.px(), trackPion.py(), trackPion.pz()};
+        ;
```

Comment:
This looks like a remnant.

Reply:
Thanks, the cut on invariant mass window disappeared for some reason. I think I had different versions of the code that overlapped for some reason, I guess some bad manipulation in git.